cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -921,138 +921,164 @@ PyMODINIT_FUNC PyInit_simulator(void)
     return simulator;
 }
 
+/* NOTE: in the following docstrings we are specifying the parameters twice, but this is necessary.
+ * The first docstring before the long '--' line specifies the __text_signature__ that is used
+ * by the help() function. And the second after the '--' line contains type annotations used by
+ * the `autodoc_docstring_signature` setting of sphinx.ext.autodoc for generating documentation
+ * because type annotations are not supported in __text_signature__.
+ */
+
 static PyMethodDef gpi_sim_hdl_methods[] = {
     {"get_signal_val_long",
         (PyCFunction)get_signal_val_long, METH_NOARGS, PyDoc_STR(
             "get_signal_val_long($self)\n"
             "--\n\n"
-            "Get the value of a signal as a long"
+            "get_signal_val_long(self) -> int\n"
+            "Get the value of a signal as an integer."
         )
     },
     {"get_signal_val_str",
         (PyCFunction)get_signal_val_str, METH_NOARGS, PyDoc_STR(
             "get_signal_val_str($self)\n"
             "--\n\n"
-            "Get the value of a signal as an ASCII string"
+            "get_signal_val_str(self) -> str\n"
```

Comment:
This is wrong I think.
```suggestion
            "get_signal_val_str(self) -> bytes\n"
```